src/Simulation/QsharpCore/ClassicalControl.cs

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using Microsoft.Quantum.Simulation.Core;
+
+namespace Microsoft.Quantum.Simulation.QuantumProcessor.Extensions
+{
+    public partial class ApplyIfElseR<__T__, __U__>
+    {
```

Comment:
The code within `GetRuntimeMetadata` below is the same across all 4 `ApplyIfElseR` variants. Is there a way to avoid code duplication here? 

I tried a prototype with a new inherited `IApplyIf` class and having each variant inherit from it, but couldn't get it to work.

Reply:
You can create a static method and have all of them call the same method.